### ISSUE TITLE
Is it possible to run this code on VS for Mac?

### ISSUE THREAD

**Issue by jaxxon33** [Azure-Samples/active-directory-b2c-dotnetcore-webapp#24]
I have the code running fine on VS for Windows but have been trying it on VS for Mac, without altering the code, with no luck. Is Windows (IIS) a requirement?

The website loads fine but below is what I see when I click on 'sign in".

<img width="537" alt="screen shot 2017-05-20 at 11 11 37 pm" src="https://cloud.githubusercontent.com/assets/1849877/26276103/fbb0741e-3db3-11e7-8a1d-a46b7a9de181.png">

It doesn't seem to redirect to the usual Microsoft login url.

<img width="291" alt="screen shot 2017-05-20 at 11 12 04 pm" src="https://cloud.githubusercontent.com/assets/1849877/26276101/f5354f9c-3db3-11e7-9df7-a085f358b853.png">

**Comment by gsacavdm** [comment]
This samples hasn't been tested on Mac yet.

Can you enable Safari's Web Inspector or use Chrome and it's Developer Tools (F12) to check out the network trace. The key question is whether it is redirecting to Azure AD B2C which immediately redirects back to localhost:8080 or whether it's just not going to Azure AD B2C at all...

Is this with the default (pre-configured) appsettings or is this only after you update them with your own?

**Comment by jaxxon33** [comment]
Thanks for your response @gsacavdm.

I see this with the original code with the default appsettings. 

One thing to note is while VS Windows runs the app on HTTPS port 44316, VS Mac uses a random HTTP port. By targeting net core 2.0, the app runs on the user specified port shown below. 

![screen shot 2017-05-21 at 10 06 04 am](https://cloud.githubusercontent.com/assets/1849877/26280196/2771d224-3e0d-11e7-80a8-dc44123f8add.png)

I am able to get it to run on port 44316 but not sure if I need to configure it to HTTPS also?

I will try your network trace suggestion later today (10AM Australian time here :))

Since VS Mac is not yet supported by this code, should I close the issue for now?

**Comment by jaxxon33** [comment]
Looks like it's going to login.microsoft.com first...

![screen shot 2017-05-21 at 2 09 45 pm](https://cloud.githubusercontent.com/assets/1849877/26281092/7e4cddc4-3e2f-11e7-889c-8f0ae1f9eb97.png)
![screen shot 2017-05-21 at 2 10 02 pm](https://cloud.githubusercontent.com/assets/1849877/26281095/8d1f2c08-3e2f-11e7-81b9-3cbb9111ae9a.png)

**Comment by jaxxon33** [comment]
Maybe more useful info in Chrome...

![screen shot 2017-05-22 at 10 29 02 am](https://cloud.githubusercontent.com/assets/1849877/26288648/bbd745d0-3ed9-11e7-84ab-ab3adf45c102.png)
![screen shot 2017-05-22 at 10 30 12 am](https://cloud.githubusercontent.com/assets/1849877/26288649/befcf476-3ed9-11e7-8fdd-11a23b6442f0.png)

**Comment by parakhj** [comment]
@jaxxon33 I tried this sample on a mac (simply downloaded it and ran it). It worked fine for me. Did you make any changes after downloading it? The one difference I notice is that by default, my app ran on port 5000.

**Comment by jaxxon33** [comment]
Hi @parakhj, as mentioned I made no changes at first. I even tried on multiple MacBooks. I have now changed to port 5000 and everything works. Thank you so much for the tip!

![screen shot 2017-05-22 at 3 50 23 pm](https://cloud.githubusercontent.com/assets/1849877/26294362/6b465a0c-3f06-11e7-90cc-05a904a005ed.png)

**Comment by jaxxon33** [comment]
I just downloaded a fresh copy of the code and ran it. By default VS for Mac doesn't run on port 5000. I changed it to port 5000 then rebuilt but my changes are ignored. I can get it to run on the port I specify by targeting netcoreapp 2.0. Then all is well except it's not running on HTTPS and so is insecure.

<img width="989" alt="screen shot 2017-05-22 at 7 24 29 pm" src="https://cloud.githubusercontent.com/assets/1849877/26301740/652176ba-3f25-11e7-8b1d-b284a85e92b9.png">
 
<img width="912" alt="screen shot 2017-05-22 at 7 29 44 pm" src="https://cloud.githubusercontent.com/assets/1849877/26301810/a186f440-3f25-11e7-8413-8baed1e6128a.png">
